rfc/rfc-92/rfc-92.md

```diff
@@ -0,0 +1,200 @@
+   * @param instant Hoodie Instant used for this transaction.
+   */
+  void completeClustering(
+      HoodieTableMetaClient metaClient,
+      FileSystemViewManager viewManager,
+      HoodieReplaceCommitMetadata replaceCommitMetadata,
+      HoodieInstant instant
+  );
+
+  /**
+   * Callback to complete finish clean operation by the plugin.
+   * @param metaClient   HoodieTableMetaClient for interacting with Hudi's internal metadata.
+   * @param viewManager  FileSystem Manager to fetch table's file-system view.
+   * @param cleanCommitMetadata Clean Commit Metadata corresponding to the clean operation.
+   * @param instant Hoodie Instant used for this transaction.
+   */
+  void completeClean(
+      HoodieTableMetaClient metaClient,
+      FileSystemViewManager viewManager,
+      HoodieCleanMetadata cleanMetadata,
+      HoodieInstant instant
+  );
+
+  /**
+   * Callback to complete finish clean operation by the plugin.
+   * @param metaClient   HoodieTableMetaClient for interacting with Hudi's internal metadata.
+   * @param viewManager  FileSystem Manager to fetch table's file-system view.
+   * @param rollbackMetadata Rollback Metadata containing the instants rolledback and files deleted.
+   * @param instant Hoodie Instant used for this transaction.
+   */
+  void completeRollback(
+      HoodieTableMetaClient metaClient,
+      FileSystemViewManager viewManager,
+      HoodieRollbackMetadata rollbackMetadata,
+      HoodieInstant instant
+  );
+
+
+  /**
+   * Callback to complete finish archive operation by the plugin.
+   * @param metaClient   HoodieTableMetaClient for interacting with Hudi's internal metadata.
+   * @param viewManager  FileSystem Manager to fetch table's file-system view.
+   * @param archivedInstants List of Hoodie Instants archived.
+   */
+  void archiveInstants(
+      HoodieTableMetaClient metaClient,
+      FileSystemViewManager viewManager,
+      List<HoodieInstant> archivedInstants
+  );
+
+  /**
+   * Return Timeline Factory which uses the external plugin's timeline state as the source of truth. 
+   * The timeline factory needs to create timelines which has instants marked complete only when the plugin's commit call succeeded.
+   */
+  TimelineFactory getTimelineFactory();
+
+  /**
+   * Provides MetadataFactory which returns metadata reader interface that uses the plugin format's metadata.
+   */
+  BaseHoodieTableMetadataFactory getMetadataFactory();
+
+  /**
+   * LockProvider corresponding to the plugin.
+   */
+  Option<LockProvider> getLockProvider(LockConfiguration lockConfiguration, StorageConfiguration storageConf);
+
+  /**
+   * Conflict resolution strategy that aligns with the plugin format.
+   */
+  Option<String> getConflictResolutionStrategyClassName(HoodieConfig config);
+}
+
+```
+## **Rollout/Adoption Plan**
+
+*   For existing tables, utility to construct the table format for the first time.
+*   Configuration of plugin to turn on specific formats. Default will be hudi.
```

Comment:
this will get confusing. There is no Hudi vs non Hudi. 

lets clearly use native vs foreign/external - to denote native table/storage format in Hudi and an external implementation backing them?

Reply:
Not sure we could render these interfaces as a standard in the lakehouse story, and not sure if there are vendors that voluntary to contribute. Technically to say, it is very hard for generalizing them to be a standard because the interface looks so Hudi specific.